lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/storage/AzureStorageUtil.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.runtime.azbatch.util.storage;
+
+import com.microsoft.windowsazure.storage.StorageException;
+import com.microsoft.windowsazure.storage.blob.*;
+import org.apache.reef.runtime.azbatch.parameters.AzureStorageContainerName;
+import org.apache.reef.tang.annotations.Parameter;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.EnumSet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Azure storage utility to upload Driver and Evaluator jars to blob storage
+ * and generate SAS URIs.
+ */
+public class AzureStorageUtil {
+  private static final Logger LOG = Logger.getLogger(AzureStorageUtil.class.getName());
+
+  private static final int BLOB_SAS_TOKEN_VALIDITY_MINUTES = 30;
+  private static final int CONTAINER_SAS_TOKEN_VALIDITY_HOURS = 24;
+
+  private final ICloudBlobClientProvider cloudBlobClientProvider;
+
+  private final String azureStorageContainerName;
```

Comment:
Would it be better to make this a parameter to the different functions? I find a utility class with an internal state variable to be a bit confusing.

Reply:
Hm, the disadvantage of that is that I'll need to litter all classes that use AzureStorageUtil with azureStorageContainerName internal state. 
Or maybe, AzureStorageUtil should be called, something like AzureStorageClient? Let me know if you feel strongly either way.